src/rdf4cpp/rdf/Literal.cpp

```diff
@@ -1556,6 +1558,48 @@ Literal Literal::substr(Literal const &start, Literal const &len, Node::NodeStor
     return this->substr(start_ix, len_ix, node_storage);
 }
 
+Literal Literal::hash_with(const char *alg, NodeStorage &node_storage) const {
+    if (!this->is_string_like())
```

Comment:
This passes for `rdf:langString` and `xsd:string` but as far as I can tell these functions are only supposed to work for `xsd:string`